FOCUS #317: BillingAccountType and SubAccountType (#287)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/columns/billingaccounttype.md

# Billing Account Type

> Billing Account Type is the label the provider uses to describe the kind of [*billing account*](#glossary:billing-account). Billing Account Type is a readable display name and not a code. Billing Account Type is commonly used for scenarios like mapping FOCUS constructs to provider constructs and summarizing costs across providers.

Author: ```suggestion
Billing Account Type is a provider-assigned name to identify the type of [*billing account*](#glossary:billing-account). Billing Account Type is a readable display name and not a code. Billing Account Type is commonly used for scenarios like mapping FOCUS constructs to provider constructs and summarizing costs across providers.
```

> The BillingAccountType column MUST be present in the billing data. This column MUST be of type String, MUST be null when [BillingAccountId](#billingaccountid) is null, and MUST NOT be null when BillingAccountId is not null. BillingAccountType MUST be a consistent, readable display value within the billing data.

Author: ```suggestion
The BillingAccountType column adheres to the following requirements:

* BillingAccountType MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports more than one possible BillingAccountType values.
* BillingAccountType MUST be of type String.
* BillingAccountType MUST NOT be null when BillingAccountId is not null.
* BillingAccountType MUST be null when BillingAccountId is null.
* BillingAccountType MUST be a consistent, readable display value.
```

## Column ID

BillingAccountType

## Display Name

Billing Account Type

## Description

> Label the provider uses to describe the kind of billing account.

Author: ```suggestion
A provider-assigned identifier for the type of *billing account* applied to the *row*.
```

## Content Constraints

| Constraint      | Value            |
| :-------------- | :--------------- |
| Column type     | Dimension        |

> | Column required | True             |

Author: ```suggestion
| Feature level   | Conditional      |
| Column required | True             |
```

| Allows nulls    | False            |
| Data type       | String           |
| Value format    | \<not specified> |

## Introduced (version)

> 1.0

Author: ```suggestion
1.2
```

# Billing Account Type

Billing Account Type is a provider-assigned name to identify the type of [*billing account*](#glossary:billing-account). Billing Account Type is a readable display name and not a code. Billing Account Type is commonly used for scenarios like mapping FOCUS constructs to provider constructs and summarizing costs across providers.

The BillingAccountType column adheres to the following requirements:

* BillingAccountType MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports more than one possible BillingAccountType value.

> * BillingAccountType MUST be of type String.

Author: ```suggestion
* BillingAccountType MUST be of type String.
* BillingAccountType MUST conform to [String Handling](#stringhandling) requirements.
```

# Billing Account Type

> Billing Account Type is a provider-assigned name to identify the type of [*billing account*](#glossary:billing-account). Billing Account Type is a readable display name and not a code. Billing Account Type is commonly used for scenarios like mapping FOCUS constructs to provider constructs and summarizing costs across providers.

Reviewer: Nit: Consider mentioning invoice reconciliation and possibly cost allocation as well...

In the case of AWS, BillingAccountType might provide information on whether the Billing Account represents the Management Account level or an underlying Cost Center, with separate invoices being issued (see [Configuring your AWS Invoices using Invoice Configuration](https://aws.amazon.com/blogs/aws-cloud-financial-management/configuring-your-aws-invoices-using-invoice-configuration/))

Author: ```suggestion
Billing Account Type is a provider-assigned name to identify the type of [*billing account*](#glossary:billing-account). Billing Account Type is a readable display name and not a code. Billing Account Type is commonly used for scenarios like mapping FOCUS and provider constructs, summarizing costs across providers, or invoicing and chargeback.
```
